index.js

```diff
@@ -205,17 +206,31 @@ deriveEq (Head);
 //. ```hs
 //. data Body a = None
 //.             | Send Any
+//.             | Json JsonValue
 //.             | Stream (Future a Readable)
 //.             | Render String Object
 //. ```
 export const Body = daggy.taggedSum ('Body', {
   None: [],
   Send: ['data'],
+  Json: ['data'],
   Stream: ['stream'],
   Render: ['template', 'data'],
 });
 
-deriveEq (Body);
+Body.prototype['fantasy-land/equals'] = function Body$FL$equals(other) {
+  return this.cata ({
+    None: _ => Body.None.is (other),
+    Send: data => Body.Send.is (other) && Z.equals (data, other.data),
+    Json: data => Body.Json.is (other) && Z.equals (data, other.data),
+    Stream: stream => Body.Stream.is (other) && stream === other.stream,
+    Render: (template, data) => (
+      Body.Render.is (other) &&
+      template === other.template &&
+      Z.equals (data, other.data)
+    ),
+  });
+};
```

Comment:
I prefer nested case folding. Let's consider the following data type:

```haskell
data Gizmo a b c = Foo a | Bar b | Baz c
```

```javascript
Gizmo.prototype['fantasy-land/equals'] = function(other) {
  return this.cata ({
    Foo: x => other.cata ({
      Foo: y => Z.equals (x, y),
      Bar: _ => false,
      Baz: _ => false,
    }),
    Bar: x => other.cata ({
      Foo: _ => false,
      Bar: y => Z.equals (x, y),
      Baz: _ => false,
    }),
    Baz: x => other.cata ({
      Foo: _ => false,
      Bar: _ => false,
      Baz: y => Z.equals (x, y),
    }),
  });
};
```

I prefer this approach because it does not rely on implementation details (the property names). The verbosity can be alleviated by defining a case-folding function that takes a default value:

```javascript
Gizmo.prototype['fantasy-land/equals'] = function(other) {
  return Gizmo.fold ({
    Foo: x => Gizmo.foldWithDefault (false) ({Foo: y => Z.equals (x, y)}),
    Bar: x => Gizmo.foldWithDefault (false) ({Bar: y => Z.equals (x, y)}),
    Baz: x => Gizmo.foldWithDefault (false) ({Baz: y => Z.equals (x, y)}),
  }) (this) (other);
};
```

This is just my preference. Relying on implementation details is usually fine in practice.

Reply:
I did it like this for the `deriveEq` function. I guess I could unify the two approaches by building on top of a `foldWithDefault` function.